doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -341,88 +338,82 @@ Rewriting with definitional equality
-*goal clauses* which consists in a list of hypotheses and, optionally,
-of a reference to the conclusion of the goal. For defined hypothesis
-it is possible to specify if the conversion should occur on the type
-part, the body part or both (default).
-
-Goal clauses are written after a conversion tactic (tactics :tacn:`set`,
-:tacn:`rewrite`, :tacn:`replace` and :tacn:`autorewrite` also use goal
-clauses) and are introduced by the keyword `in`. If no goal clause is
-provided, the default is to perform the conversion only in the
-conclusion.
-
-For backward compatibility, the notation :n:`in {+ @ident}` performs
-the conversion in hypotheses :n:`{+ @ident}`.
-
-.. tacn:: cbv {? @strategy_flag }
-          lazy {? @strategy_flag }
-   :name: cbv; lazy
-
-   These parameterized reduction tactics apply to any goal and perform
-   the normalization of the goal according to the specified flags. In
-   correspondence with the kinds of reduction considered in Coq namely
-   :math:`\beta` (reduction of functional application), :math:`\delta`
-   (unfolding of transparent constants, see :ref:`vernac-controlling-the-reduction-strategies`),
-   :math:`\iota` (reduction of
-   pattern matching over a constructed term, and unfolding of :g:`fix` and
-   :g:`cofix` expressions) and :math:`\zeta` (contraction of local definitions), the
-   flags are either ``beta``, ``delta``, ``match``, ``fix``, ``cofix``,
-   ``iota`` or ``zeta``. The ``iota`` flag is a shorthand for ``match``, ``fix``
-   and ``cofix``. The ``delta`` flag itself can be refined into
-   :n:`delta [ {+ @qualid} ]` or :n:`delta - [ {+ @qualid} ]`, restricting in the first
-   case the constants to unfold to the constants listed, and restricting in the
-   second case the constant to unfold to all but the ones explicitly mentioned.
-   Notice that the ``delta`` flag does not apply to variables bound by a let-in
-   construction inside the :n:`@term` itself (use here the ``zeta`` flag). In
-   any cases, opaque constants are not unfolded (see :ref:`vernac-controlling-the-reduction-strategies`).
-
-   Normalization according to the flags is done by first evaluating the
+   .. prodn::
+      reductions ::= {+ @reduction }
+      | @delta_reductions
+      reduction ::= beta
+      | iota
+      | match
+      | fix
+      | cofix
+      | zeta
+      | delta {? @delta_reductions }
+      delta_reductions ::= {? - } [ {+ @reference } ]
+
+   Normalize the goal as specified by :n:`@reductions`. The reductions include:
+
+   `beta`
+     :term:`beta-reduction` of functional application
+
+   :n:`delta {? @delta_reductions }`
+     :term:`delta-reduction`: unfolding of transparent constants, see :ref:`controlling-the-reduction-strategies`.
+     The form with the keyword `delta` indicates only the `delta` reduction, while the form
+     without the keyword in :n:`@reductions` includes `beta`, `iota` and `zeta`
+     reductions in addition to `delta`.
```

Comment:
This can seem a bit odd to the reader, whereas it wouldn't be so odd if it was specified above (rather than below) that if no reduction rule is specified explicitly, then they are all applied. Then, it could be written that using `delta_reductions` without an explicit `delta` amounts to fine-tuning the delta rule while keeping all the other reductions on.

Reply:
like this?

Normalize the goal as specified by :n:`@reductions`.  If no reductions are specified by name, all reductions are applied.  If any reductions are specified by name, then only the named reductions are applied.  The reductions include:
   :
   :n:`delta {? @delta_reductions }`
     :term:`delta-reduction`: unfolding of transparent constants, see :ref:`controlling-the-reduction-strategies`.  The form in :n:`@reductions` without the keyword `delta` includes `beta`,  `iota` and `zeta` reductions in addition to `delta` using the given :n:`@delta_reductions`.

Reply:
Yes, LGTM!